schematics/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, OnInit, ViewChild, <% if(!!viewEncapsulation) { %>, ViewEncapsulation<% }%><% if(changeDetection !== 'Default') { %>, ChangeDetectionStrategy<% }%> } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators/map';
+import { MatPaginator, MatSort } from '@angular/material';
+import { DataSource } from '@angular/cdk/collections';
+import { of } from 'rxjs/observable/of';
+import { merge } from 'rxjs/observable/merge';
+
+/** Todo: Replace this with your own data model type */
+export interface DataItem {
+  name: string;
+  id: number;
+}
+
+function compare(a, b, isAsc) {
+  return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
+}
+
+export class ExampleDataSource extends DataSource<any> {
+  data: DataItem[] = [
```

Comment:
I would move this array to be a constant at the top like `EXAMPLE_DATA`, and then when it's assigned in the class it should have a comment like
```
// TODO: replace this with real data from your application
```